client.go

```diff
@@ -166,13 +167,21 @@ func parseResponseBody(rsp *http.Response) ([]byte, error) {
 	}
 	defer rsp.Body.Close()
 	body, err := ioutil.ReadAll(rsp.Body)
+	if err != nil {
+		return nil, err
+	}
 
 	if rsp.StatusCode >= 400 {
-		re := responseError{}
-		if err = json.Unmarshal(body, &re); err != nil {
-			return nil, errors.WithStack(fmt.Errorf("response error: %d. json error: %v", rsp.StatusCode, err))
-		}
-		return nil, errors.WithStack(fmt.Errorf("response error: %d, %s. %s", rsp.StatusCode, re.Code, re.Message))
+		return nil, maxcomputeResponseError(rsp.StatusCode, body)
 	}
 	return body, err
 }
+
+func maxcomputeResponseError(statusCode int, body []byte) error {
```

Comment:
I feel `maxcompute` is redundant...

Reply:
`responseError` is a `struct` name here already

Reply:
How about `parseResponseError`?

Reply:
Agree